### PR TITLE
Refactor extract meca to free up memory

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: src/activities/extract-meca.ts [hunk 1]
```diff
@@ -47,21 +47,6 @@ type ManifestItem = {
...
-  mkdirSync(outputDir, { recursive: true });
-  writeFileSync(outputPath, content, 'base64');
-  return {
-    ...item,
-    localPath: outputPath,
-  };
-};
-
 export const extractMeca = async (version: VersionedReviewedPreprint): Promise<MecaFiles> => {
   const tmpDirectory = await mkdtemp(`${tmpdir()}/epp_content`);
```
Comment (scottaubrey) [inline src/activities/extract-meca.ts:51]: I know this was pre-existing, but this surely means that we have the potential to end up with files overwriting each other. 

What do you think to 
```
const tmpDirectory = await mkdtemp(`${tmpdir()}/epp_content/${version.id}`);
``` 
?

File: src/activities/extract-meca.ts [hunk 2]
```diff
@@ -73,21 +58,42 @@ export const extractMeca = async (version: VersionedReviewedPreprint): Promise<M
     Bucket: source.Bucket,
     Key: source.Key,
   };
-  const buffer = await s3.send(new GetObjectCommand(getObjectCommandInput))
-    .then((obj) => obj.Body?.transformToByteArray());
+  const data = await s3.send(new GetObjectCommand(getObjectCommandInput));
+
+  const downloadMeca = (body: Readable, localPath: string) => new Promise((resolve, reject) => {
```
Comment (scottaubrey) [inline src/activities/extract-meca.ts:63]: Notpick: This isn't very specific to a Meca file - should we move to a util (with a new name - maybe `downloadS3File` or something?)

**Review by scottaubrey — CHANGES_REQUESTED** [review]
One tiny thing not even to do with your changes, and a small suggestion, but otherwise it looks good to me

File: src/activities/extract-meca.ts (hunk 1 above)
Comment (scottaubrey) [reply src/activities/extract-meca.ts:51]: Actually, that's nonsense.

**Review by scottaubrey — APPROVED** [review]
Sorry, updated the review after realising my misunderstanding. I'd like the download function moved out, but otherwise I think it's good to go.

File: src/activities/extract-meca.ts (hunk 2 above)
Comment (nlisgo) [reply src/activities/extract-meca.ts:63]: I will address this refactor this morning.

Thanks for checking it over.

Comment (nlisgo) [reply src/activities/extract-meca.ts]: I have moved this to S3Bucket and renamed to `streamToFile`. I have tried to add a test but it's creating a lot of noise unrelated to adding a test just for this function. We should prioritise tests for this and other functions at another time.

**Review by scottaubrey — APPROVED** [review]

**Pull request merged**